### PR TITLE
session: disable telemetry when using unistore

### LINKED ISSUE #34314: Annoying telemetry log when using unistore
## Enhancement
image

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #34314

Problem Summary:
Annoying telemetry log when using unistore
### What is changed and how it works?
Enable telemetry only when the store is TiKV, then its etcd client is not nil.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Start a TiDB with unistore.
No annoying log show.
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -2839,8 +2839,10 @@ func BootstrapSession(store kv.Storage) (*domain.Domain, error) {
 		return nil, err
 	}
 
-	dom.TelemetryReportLoop(ses[5])
-	dom.TelemetryRotateSubWindowLoop(ses[5])
+	if store.Name() == "TiKV" {
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply session/session.go:2842]: I think so.

**Review by AilinKid — APPROVED** [review]

File: session/session.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (crazycs520) [reply session/session.go:2842]: maybe define a const var is better？

File: session/session.go [hunk 2]
```diff
@@ -2839,8 +2839,10 @@ func BootstrapSession(store kv.Storage) (*domain.Domain, error) {
...
+		dom.TelemetryReportLoop(ses[5])
```
Comment (breeswish) [inline session/session.go:2843]:
```suggestion
	if store.Name() == "TiKV" {
		// We only want telemetry data in production-like clusters. When TiDB is deployed over other engines, for example, unistore engine (used for local tests), we just skip it.
		dom.TelemetryReportLoop(ses[5])
```

**Review by breeswish — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**